JS/Lesson18/onur/recursion.js

```diff
@@ -53,39 +56,68 @@ function printArrLoop(array) {
 }
 printArr(arr);
 printArrLoop(arr);
+
+console.log('----------------------------');
 // 2. Write a recursive function to capitalize the first letter of each word in a sentence.
-const ex2sentence =
-	'a recursive function to capitalize the first letter of each word in a sentence.  ';
-function toCapitalizeEachWord(givenString) {
-	if (typeof givenString !== 'string') {
-		console.log('Invalid Type or input');
-		return '';
-	} else if (givenString === ''){ // this should be a separate case so we do not log 'Invalid Type or input' message after capitalazing all the words in the sentense.
-    return '';
-  } else {
-		const indexOfFirstSpace = givenString.indexOf(' ');
-		const firstLetter = givenString[0].toUpperCase();
-		let restOfFirstWord,
-			restOfTheSentense = '';
-
-		if (indexOfFirstSpace === -1) {
-			restOfFirstWord = givenString.slice(1);
-		} else {
-			// 'afaasdas sfdsds'
-			const firstWord = givenString.slice(0, indexOfFirstSpace); // afaasdas
-			const restOfSentence = givenString.slice(indexOfFirstSpace + 1); // sfdsds
+// const ex2sentence =
+// 	'a recursive function to capitalize the first letter of each word in a sentence.  ';
+// function toCapitalizeEachWord(givenString) {
+// 	if (typeof givenString !== 'string') {
+// 		console.log('Invalid Type or input');
+// 		return '';
+// 	} else if (givenString === '') { // this should be a separate case so we do not log 'Invalid Type or input' message after capitalazing all the words in the sentense.
+// 		return '';
+// 	} else {
+// 		const indexOfFirstSpace = givenString.indexOf(' ');
+// 		const firstLetter = givenString[0].toUpperCase();
+// 		let restOfFirstWord,
+// 			restOfTheSentense = '';
 
-			restOfFirstWord = firstWord.slice(1);
-			restOfTheSentense = toCapitalizeEachWord(restOfSentence);
-		}
-		return firstLetter + restOfFirstWord + ' ' + restOfTheSentense;
+// 		if (indexOfFirstSpace === -1) {
+// 			restOfFirstWord = givenString.slice(1);
+// 		} else {
+// 			// 'afaasdas sfdsds'
+// 			const firstWord = givenString.slice(0, indexOfFirstSpace); // afaasdas
+// 			const restOfSentence = givenString.slice(indexOfFirstSpace + 1); // sfdsds
+
+// 			restOfFirstWord = firstWord.slice(1);
+// 			restOfTheSentense = toCapitalizeEachWord(restOfSentence);
+// 		}
+// 		return firstLetter + restOfFirstWord + ' ' + restOfTheSentense;
+// 	}
+// }
+
+// console.log(toCapitalizeEachWord('a'));
+// console.log(toCapitalizeEachWord('afaasdas sfdsds sfasf asdsa'));
+// console.log(toCapitalizeEachWord(ex2sentence));
+
+// I also did it with a different way. ***
+
+const ex2sentence = 'a recursive function to capitalize the first letter of each word in a sentence.';
+
+function capitalizeFirstLetter(sentence) {
+	if (sentence === '') {
+		return '';
+	} else if (typeof sentence !== 'string') {
+		console.log("Invalid input");
 	}
+
+	const TotalWords = sentence.split(' ');
```

Comment:
```suggestion
	const totalWords = sentence.split(' ');
```
In JS we usually use camelCase to name variables :)